packages/react/.changeset/little-bikes-dress.md

```diff
@@ -0,0 +1,27 @@
+---
+"@gadgetinc/react": minor
+---
+
+# Added support for `upsert` actions in AutoForm.
+
+- Models that have `create` and `update` actions get the `upsert` meta-action
```

Comment:
I'm not 100% sure this usage makes sense to me. I have to admit I can still get fuzzy on upsert sometimes, but I think what we want is something along the lines of: 

"given some attribute that I will always submit with my upsert action -- lets say, "email", or "shopifyProductId", either create the record in the gadget db, or update the existing record"

Reply:
That `on` parameter based stuff is currently not supported in AutoForm. Since that parameter is an array and array params are an open question right now, I'd prefer to revisit this note if we add support for array custom action params